#! /bin/sh /usr/share/dpatch/dpatch-run
## 03forcecleanup.dpatch by  <az@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix for #572792: we force a extra-clean cleanup for all remove* ops

@DPATCH@
diff -urNad duplicity-0.6.08b~/duplicity duplicity-0.6.08b/duplicity
--- duplicity-0.6.08b~/duplicity	2010-03-20 14:15:52.000000000 +1000
+++ duplicity-0.6.08b/duplicity	2010-03-20 14:15:52.945258047 +1000
@@ -768,12 +768,24 @@
             for set in setlist:
                 set.delete()
             col_stats.set_values(sig_chain_warning=None)
+
+            # force a cleanup operation to get rid of unnecessary old cruft
+            # we said we want to remove them! didn't we, huh?
+            # bad duplicity, bad doggy!
+            # note: in the long run backing out changeset 616 might be
+            # better, but for now this will ease the pain.
+            globals.extra_clean=True
+            cleanup(col_stats)
     else:
         log.Notice(gettext.ngettext("Found old backup set at the following time:",
                                     "Found old backup sets at the following times:",
                                     len(setlist)) +
                    "\n" + set_times_str(setlist) + "\n" +
                    _("Rerun command with --force option to actually delete."))
+        # see above for rationale.
+        # this here is to print a list of to-be-removed files (--force is off)
+        globals.extra_clean=True
+        cleanup(col_stats)
 
 
 def sync_archive():
diff -urNad duplicity-0.6.08b~/duplicity.1 duplicity-0.6.08b/duplicity.1
--- duplicity-0.6.08b~/duplicity.1	2010-03-12 11:39:07.000000000 +1000
+++ duplicity-0.6.08b/duplicity.1	2010-03-20 14:24:42.856263504 +1000
@@ -184,6 +184,14 @@
 .I --force
 will be needed to delete the files rather than just list them.
 
+.B Note: 
+the Debian version of duplicity automatically runs a 
+cleanup --extra-clean whenever old backup sets are removed (i.e. if one
+of the remove commands is run with the --force option present and 
+if something removable is found). This is to 
+limit the amount of old outdated material that otherwise accumulates 
+in the archive dir.
+
 .TP
 .BI "remove-all-but-n-full " count
 Delete all backups sets that are older than the count:th last full
@@ -196,6 +204,9 @@
 .I --force
 will be needed to delete the files rather than just list them.
 
+The note regarding automatic cleanups above
+also applies to remove-all-but-n-full.
+
 .TP
 .B verify
 Enter verify mode instead of restore.  If the --file-to-restore option
diff -urNad duplicity-0.6.08b~/src/collections.py duplicity-0.6.08b/src/collections.py
--- duplicity-0.6.08b~/src/collections.py	2010-03-20 14:15:52.000000000 +1000
+++ duplicity-0.6.08b/src/collections.py	2010-03-20 14:15:52.945258047 +1000
@@ -991,8 +991,6 @@
             if self.matched_chain_pair:
                 matched_sig_chain = self.matched_chain_pair[0]
                 for sig_chain in self.all_sig_chains:
-                    print sig_chain.start_time, matched_sig_chain.start_time, 
-                    print sig_chain.end_time, matched_sig_chain.end_time
                     if (sig_chain.start_time == matched_sig_chain.start_time and
                         sig_chain.end_time == matched_sig_chain.end_time):
                         old_sig_chains.remove(sig_chain)
